src/configuration.py

```diff
@@ -35,15 +35,15 @@ def set_val(self, v):
             if isinstance(v, self.type) or (self.is_optional and v is None):
                 self.val = v
             else:
-                raise TypeError
+                raise TypeError(f"{v} is not of type {self.type}")
```

Comment:
Treść może być myląca dla kogoś kto czyta. bo wyświetli się np "4 is not type integer" ale przecież wygląda że jest. piwinno być racaej coś w stylu "otrzymany parametr jest typu x a wymagane jest y"